sonarr/qualitydefinition.go

```diff
@@ -29,52 +30,55 @@ func (s *Sonarr) GetQualityDefinitions() ([]*QualityDefinition, error) {
 	return s.GetQualityDefinitionsContext(context.Background())
 }
 
+// GetQualityDefinitionsContext returns all configured quality definitions.
 func (s *Sonarr) GetQualityDefinitionsContext(ctx context.Context) ([]*QualityDefinition, error) {
 	var output []*QualityDefinition
 
 	if err := s.GetInto(ctx, bpQualityDefinition, nil, &output); err != nil {
-		return nil, fmt.Errorf("api.Get(qualityDefinition): %w", err)
+		return nil, fmt.Errorf("api.Get(%s): %w", bpQualityDefinition, err)
 	}
 
 	return output, nil
 }
 
 // GetQualityDefinition returns a single quality definition.
-func (s *Sonarr) GetQualityDefinition(qualityDefinitionID int) (*QualityDefinition, error) {
+func (s *Sonarr) GetQualityDefinition(qualityDefinitionID int64) (*QualityDefinition, error) {
 	return s.GetQualityDefinitionContext(context.Background(), qualityDefinitionID)
 }
 
-func (s *Sonarr) GetQualityDefinitionContext(ctx context.Context, qualityDefinitionID int) (*QualityDefinition, error) {
-	var output *QualityDefinition
+// GetQualityDefinitionContext returns a single quality definition.
+func (s *Sonarr) GetQualityDefinitionContext(ctx context.Context, qdID int64) (*QualityDefinition, error) {
+	var output QualityDefinition
 
-	uri := path.Join(bpQualityDefinition, strconv.Itoa(qualityDefinitionID))
+	uri := path.Join(bpQualityDefinition, strconv.FormatInt(qdID, starr.Base10))
 	if err := s.GetInto(ctx, uri, nil, &output); err != nil {
-		return nil, fmt.Errorf("api.Get(qualityDefinition): %w", err)
+		return nil, fmt.Errorf("api.Get(%s): %w", uri, err)
 	}
 
-	return output, nil
+	return &output, nil
 }
 
-// UpdateQualityDefinition updates the quality definition.
-func (s *Sonarr) UpdateQualityDefinition(definition *QualityDefinition) (*QualityDefinition, error) {
-	return s.UpdateQualityDefinitionContext(context.Background(), definition)
+// UpdateQualityDefinition updates all quality definitions.
+func (s *Sonarr) UpdateQualityDefinitions(definitions []*QualityDefinition) ([]*QualityDefinition, error) {
+	return s.UpdateQualityDefinitionsContext(context.Background(), definitions)
```

Comment:
please do not change this. If you want to modify them all together, add another method instead

Reply:
good call!